src/main/java/org/beanpod/switchboard/controller/EncoderController.java

```diff
@@ -110,18 +111,19 @@
     return ResponseEntity.ok("Encoder with serial number " + serialNumber + " Deleted");
   }
 
-  @PutMapping
-  public ResponseEntity<EncoderDto> updateEncoder(@RequestBody EncoderDto encoderDto) {
+  @Override
+  public ResponseEntity<EncoderModel> updateEncoder(@Valid EncoderModel encoderModel) {
     UserEntity user = userDao.findUser(request.getUserPrincipal().getName());
 
-    Optional<EncoderDto> encoder = encoderDao.findEncoder(user, encoderDto.getSerialNumber());
-    if (encoder.isEmpty()) {
-      throw new ExceptionType.DeviceNotFoundException(encoderDto.getSerialNumber());
-    }
-    return ResponseEntity.ok(encoderDao.save(user, encoderDto));
+    return encoderDao
+        .findEncoder(user, encoderModel.getSerialNumber())
+        .map(encoderDto -> encoderDao.save(user, encoderDto))
+        .map(encoderMapper::toEncoderModel)
+        .map(ResponseEntity::ok)
+        .orElseThrow(this::getUnknownException);
```

Comment:
getting a 400 error updating an encoder

Reply:
![image](https://user-images.githubusercontent.com/41707836/114258229-1d8db300-9993-11eb-8cc4-948323c66d2a.png)
does it have to do with my date format?

Reply:
yes it ended up being the format

Reply:
However, using the right format, the date does not get update
![image](https://user-images.githubusercontent.com/41707836/114258249-4e6de800-9993-11eb-8bb7-d5ac502d432d.png)

Reply:
it was the date format, i fixed that but now if i try to update the date, it doesnt get changed

Reply:
That was a bug, good catch!